go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -401,13 +420,32 @@ func (c *Controller) syncService(key string) error {
 }
 
 // RequestFullSync re-syncs every service that currently exists
-func (c *Controller) RequestFullSync() {
+func (c *Controller) RequestFullSync(nodeIPv4Template, nodeIPv6Template *Template) {
```

Comment:
> So what I would do here is:
> 
>     1. pass a copy of the nodeInfos as well
> 
>     2. save all of it in Controller variables (with appropriate mutex synchronization, maybe the alreadyAppliedLock can be repurposed for this)
> 
>     3. Use those variables instead of `c.nodeTracker.allNodes()` when synchronizing services
> 

Sure, but why?  What's wrong with passing a deep copy of the templates.  We have eventual consistency and we don't have to care about getting the synchronization right?  Also, resync is not a common operation.

> 
> One additional quesiotn here: is it possible that only template values have changed in a way that a sync would not be required?

Yes, it's possible.  If a node changes IP, for example.  But I don't think I want to take care of that partial update through this PR.  We can open an issue to do a follow up.  Even without templates, a node IP change currently triggers a full resync of services.

Reply:
And hopefully we don't have an issue by not copying nodeInfos, but that is how it already was before this PR.

Reply:
OK, I kept it as is for now.